src/status_im/ui/screens/discover/events.cljs

```diff
@@ -1,148 +1,170 @@
+                            :contacts           contacts
+                            :message-id         random-id}
+     :dispatch-later [{:ms       (* request-discoveries-interval-s 1000)
+                       :dispatch [:request-discoveries]}]}))
+
+(handlers/register-handler-fx
+  :discoveries-send-portions
+  (fn [{{:keys [current-public-key web3]
+         :contacts/keys [contacts]} :db}
+       [_ to]]
+    {::send-portions {:current-public-key current-public-key
+                      :web3 web3
+                      :contacts contacts
+                      :to to}}))
+
+(handlers/register-handler-fx
+  :discoveries-request-received
+  (fn [{{:keys [current-public-key web3]
+         :contacts/keys [contacts]} :db}
+       [_ {:keys [from]}]]
+    {::send-portions {:current-public-key current-public-key
+                      :web3 web3
+                      :contacts contacts
+                      :to from}}))
+
+(handlers/register-handler-fx
+  :discoveries-response-received
+  [(re-frame/inject-cofx :now)]
+  (fn [{{:keys [discoveries]
+         :contacts/keys [contacts]} :db
+        now                         :now}
+       [_ {:keys [payload from]}]]
+    (when (get contacts from)
+      (when-let [data (:data payload)]
+        (doseq [{:keys [message-id] :as discover} data]
+          (when (and (not (discoveries/exists? message-id))
+                     (not (get discoveries message-id)))
+            (let [discover (assoc discover :created-at now)]
+
+              {:dispatch [:add-discover discover]})))))))
+
+(handlers/register-handler-fx
+  :status-received
+  [(re-frame/inject-cofx :now)]
+  (fn [{{:keys [discoveries]} :db
+        now                   :now}
+       [_ {:keys [from payload]}]]
+    (when (and (not (discoveries/exists? (:message-id payload)))
+               (not (get discoveries (:message-id payload))))
+      (let [{:keys [message-id status hashtags profile]} payload
+            {:keys [name profile-image]} profile
+            discover {:message-id   message-id
+                      :name         name
+                      :photo-path   profile-image
+                      :status       status
+                      :whisper-id   from
+                      :tags         (map #(hash-map :name %) hashtags)
+                      :created-at   now}]
+        {:dispatch [:add-discover discover]}))))
+
+(handlers/register-handler-fx
+  :reload-tags
+  (fn [{:keys [db]} _]
+    {:db (assoc db
+                :tags (discoveries/get-all-tags)
+                :discoveries (reduce (fn [acc {:keys [message-id] :as discover}]
+                                       (assoc acc message-id discover))
+                                     {}
+                                     (discoveries/get-all :desc)))}))
+
+(handlers/register-handler-fx
+  :add-discover
+  (fn [{:keys [db]} [_ discover]]
+    {::save-discover discover
+     :db (assoc db :new-discover discover)
+     :dispatch [:reload-tags]}))
```

Comment:
The same as above, I see no reason why we need to dispatch `:reload-tags` as an distinct event. 
When it's because of the fact that `discover` added here has to be already in realm in order to pick-it up in `:reload-tags`, then it's very dangerous assumption to made (order of effects realisation is not guaranteed in `re-frame`) and we should express that flow explicitly for example by passing callback event creator to `::save-discover` effect.